File: Level of Nodes[19-Oct-2023]/Solution.cpp
//  >> CPP CODE <<

class Solution
{
	public:
	//Function to find the level of node X.
	int nodeLevel(int V, vector<int> adj[], int X) {
    int level = 0;
    queue<int> q;
    vector<bool> visited(V, false);

    q.push(0);

    while (!q.empty()) {
        int size = q.size();

        for (int i = 0; i < size; i++) {
            int nn = q.front();
            q.pop();
            visited[nn] = true;

            if (nn == X)
                return level;

            for (int neighbor : adj[nn]) {
                if (!visited[neighbor])
                    q.push(neighbor);
            }
        }

        level++;
    }

    return -1;
}
};

File: Level of Nodes[19-Oct-2023]/Solution.java
//   >> JAVA CODE <<

class Solution
{
    //Function to find the level of node X.
    int nodeLevel(int V, ArrayList<ArrayList<Integer>> adj, int X)
    {
        int level=0;
        
        Queue<Integer> q=new LinkedList<>();
        boolean[] visited=new boolean[V];
        
        q.add(0);
        
        while(!q.isEmpty())
        {
            int size=q.size();
            
            for(int i=0;i<size;i++)
            {
                int nn=q.poll();
                visited[nn]=true;
                
                if(nn==X) return level;
                
                for(var neighbor:adj.get(nn))
                {
                    if(visited[neighbor]==false)
                        q.add(neighbor);
                }
            }
            
            level++;
        }
        
        return -1;
    }
}